src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Parser.cs

```diff
@@ -144,11 +144,43 @@ public IReadOnlyList<LoggerClass> GetLogClasses(IEnumerable<ClassDeclarationSynt
                                             }
 
                                             ImmutableArray<TypedConstant> items = attributeData.ConstructorArguments;
-                                            Debug.Assert(items.Length == 3);
 
-                                            eventId = items[0].IsNull ? -1 : (int)GetItem(items[0]);
-                                            level = items[1].IsNull ? null : (int?)GetItem(items[1]);
-                                            message = items[2].IsNull ? string.Empty : (string)GetItem(items[2]);
+                                            switch (items.Length)
```

Comment:
Do we correctly handle, e.g.
```C#
[LoggerMessage(level: LogLevel.Warning, message: "custom message", eventId: 0)]
```
?

Reply:
I confirmed this work with and without the new changes.